jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ExtraDirectoriesParameters.java

```diff
@@ -67,16 +71,22 @@ public ExtraDirectoryParameters(Project project) {
 
   /**
    * Sets paths. {@code paths} can be any suitable object describing file paths convertible by
-   * {@link Project#files} (such as {@code List<File>}).
+   * {@link Project#files} (such as {@link File}, {@code List<File>}, or {@code List<String>}).
    *
    * @param paths paths to set.
    */
-  // non-plural to retain backward-compatibility for the "jib.extraDirectory.path" config parameter
-  public void setPath(Object paths) {
+  public void setPaths(Object paths) {
+    jibExtension.extraDirectoriesConfigured = true;
```

Comment:
I'm curious how necessary this is, would prefer we didn't need to keep track of things with flags?

Reply:
Yeah, this is quite unfortunate. The Gradle case wasn't easy for a few reasons.

1. I couldn't find a way (if at all) to log inside `JibExtension` or `ExtraDirectoriesParameters`.
1. The way we can extract `extraDirectories.paths` or `extraDirectory.path` in our codebase for consumption is to first go through `getExtraDirectories()`, which is also used by Gradle to set in config values, so unlike Maven, it's not possible to check if the user is using the old or new syntax (or both) inside `getExtraDirectories()`. On top of that, Gradle always calls both `getExtraDirectory()` and `getExtraDirectories()` regardless of whether the user has ever configured anything.

So, although this is very unfortunate, I couldn't find a good alternative.